### PR TITLE
gh-124498: Fix `TypeAliasType` not to be generic, when `type_params=()`

### PR DESCRIPTION
**Pull request by sobolevn**
* Issue: gh-124498

### CODE REVIEW

File: Lib/test/test_type_aliases.py [hunk 1]
```diff
@@ -211,6 +211,19 @@ def test_generic(self):
         self.assertEqual(TA.__value__, list[T])
         self.assertEqual(TA.__type_params__, (T,))
         self.assertEqual(TA.__module__, __name__)
+        self.assertIs(TA[int].__class__, types.GenericAlias)
```
Comment (JelleZijlstra) [inline Lib/test/test_type_aliases.py]:
```suggestion
        self.assertIs(type(TA[int]), types.GenericAlias)
```
The `__class__` attribute may be a lie, let's check the real type.

File: Objects/typevarobject.c [hunk 2]
```diff
@@ -1957,7 +1957,8 @@ typealias_reduce_impl(typealiasobject *self)
 static PyObject *
 typealias_subscript(PyObject *self, PyObject *args)
 {
-    if (((typealiasobject *)self)->type_params == NULL) {
+    typealiasobject *ta = ((typealiasobject *)self);
+    if (ta->type_params == NULL || PyTuple_GET_SIZE(ta->type_params) == 0) {
```
Comment (JelleZijlstra) [inline Objects/typevarobject.c]: What do you think of instead changing `typealias_new_impl` so it replaces an empty tuple with NULL? That has the advantage that we don't have two different possible representations for an equivalent type alias object.

Comment (sobolevn) [reply Objects/typevarobject.c]: Good idea!

File: Objects/typevarobject.c [hunk 3]
```diff
@@ -1914,7 +1914,16 @@ typealias_alloc(PyObject *name, PyObject *type_params, PyObject *compute_value,
         return NULL;
     }
     ta->name = Py_NewRef(name);
-    ta->type_params = Py_IsNone(type_params) ? NULL : Py_XNewRef(type_params);
+    if (
+        type_params == NULL
```
Comment (JelleZijlstra) [inline Objects/typevarobject.c:1918]: Can it ever be NULL? Can't hurt to have this check though.

**Review by JelleZijlstra — APPROVED** [review]

File: Objects/typevarobject.c (hunk 3 above)
Comment (sobolevn) [reply Objects/typevarobject.c:1918]: Old code assumed this path with `Py_XNewRef`, so I just followed the same style.

**Pull request merged**